pallets/gear-program/src/lib.rs

```diff
@@ -278,23 +279,26 @@ pub mod pallet {
         type OriginalCodeStorage = OriginalCodeStorageWrap<T>;
     }
 
-    impl<Runtime: Config> common::ProgramStorage for pallet::Pallet<Runtime> {
-        type InternalError = Error<Runtime>;
+    impl<T: Config> common::ProgramStorage for pallet::Pallet<T> {
+        type InternalError = Error<T>;
         type Error = DispatchError;
+        type BlockNumber = T::BlockNumber;
 
-        type ProgramMap = ProgramStorageWrap<Runtime>;
-        type MemoryPageMap = MemoryPageStorageWrap<Runtime>;
-        type WaitingInitMap = WaitingInitStorageWrap<Runtime>;
+        type ProgramMap = ProgramStorageWrap<T>;
+        type MemoryPageMap = MemoryPageStorageWrap<T>;
+        type WaitingInitMap = WaitingInitStorageWrap<T>;
 
         fn pages_final_prefix() -> [u8; 32] {
-            MemoryPageStorage::<Runtime>::final_prefix()
+            MemoryPageStorage::<T>::final_prefix()
         }
     }
 
     #[cfg(feature = "debug-mode")]
-    impl<Runtime: Config> IterableMap<(ProgramId, Program)> for pallet::Pallet<Runtime> {
-        type DrainIter = PrefixIterator<(ProgramId, Program)>;
-        type Iter = PrefixIterator<(ProgramId, Program)>;
+    impl<Runtime: Config> IterableMap<(ProgramId, (Program, Runtime::BlockNumber))>
```

Comment:
Replace "Runtime" with "T", please for the debug impl also

Reply:
done